cmake/Modules/FindComputeCpp.cmake

```diff
@@ -370,23 +414,48 @@ endfunction()
 #  binaryDir : Intermediate directory to output the integration header.
 #  sourceFiles : Source files to be compiled for SYCL.
 #
-function(add_sycl_to_target targetName binaryDir sourceFiles)
+function(add_sycl_to_target)
+  set(options)
+  set(oneValueArgs TARGET BINARY_DIR)
+  set(multiValueArgs SOURCES)
+  cmake_parse_arguments(add_sycl_to_target "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(targetName ${add_sycl_to_target_TARGET})
+  set(binaryDir ${add_sycl_to_target_BINARY_DIR})
+  set(sourceFiles ${add_sycl_to_target_SOURCES})
 
-  set(sourceFiles ${sourceFiles} ${ARGN})
   set(fileCounter 0)
```

Comment:
The filecounter is only used when not using the driver. This could be moved down into the `else` branch below.